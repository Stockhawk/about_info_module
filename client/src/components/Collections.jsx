import React from 'react';
import { Popup, TransitionablePortal } from 'semantic-ui-react';
import Hover from './Hover.jsx';

class Collections extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      hover: true,
      hoveredTag: [],
    };
    this.handleMouseOver = this.handleMouseOver.bind(this);
    this.onHoverGetValue = this.onHoverGetValue.bind(this);
  }


  onHoverGetValue(e) {
    const name = e.target.innerHTML;

    let tagHovered;
    this.props.tag.forEach((tag) => {
      if (tag.tag === name) {
        console.log(tag)
        tagHovered = tag;
        return;
      }
    });
    this.setState({
      hoveredTag: tagHovered,
    });
  }

  toggleHoverState() {
    return {
      hover: !this.state.hover,
    };
  }

  handleMouseOver() {
    this.setState(this.toggleHoverState);
  }

  render() {
    return (
      <div className="collections">
        <div id="collections">
          Collections
        </div>
        <br></br>
        {this.props.stockInfo.map(stock => (
          <div key={stock.symbol}>
            <div>
              <Popup
                trigger={<div className="tags" onMouseEnter={e => this.onHoverGetValue(e)}>{stock.tags[0]}</div>}
                content={<Hover tags={this.state.hoveredTag} />}
                hoverable
                position="top center"
              />
              &nbsp;
              <Popup
                trigger={<div className="tags" onMouseEnter={e => this.onHoverGetValue(e)}>{stock.tags[1]}</div>}
                content={<Hover tags={this.state.hoveredTag} />}
                hoverable
                position="top center"
              />
              &nbsp;
              <Popup
                trigger={<div className="tags" onMouseEnter={e => this.onHoverGetValue(e)}>{stock.tags[2]}</div>}
                content={<Hover tags={this.state.hoveredTag} />}
                hoverable
                position="top center"
              />
              &nbsp;
              <Popup
                trigger={<div className="tags" onMouseEnter={e => this.onHoverGetValue(e)}>{stock.tags[3]}</div>}
                content={<Hover tags={this.state.hoveredTag} />}
                hoverable
                position="top center"
              />
            </div>
            <div>
              <Popup
                trigger={<div id="tags2" onMouseEnter={e => this.onHoverGetValue(e)}>{stock.tags[4]}</div>}
                content={<Hover tags={this.state.hoveredTag} />}
                hoverable
                position="top center"
              />
              &nbsp;
              <Popup
                trigger={<div id="tags2" onMouseEnter={e => this.onHoverGetValue(e)}>{stock.tags[5]}</div>}
                content={<Hover tags={this.state.hoveredTag} />}
                hoverable
                position="top center"
              />
              &nbsp;
              <Popup
                trigger={<div id="tags2" onMouseEnter={e => this.onHoverGetValue(e)}>{stock.tags[6]}</div>}
                content={<Hover tags={this.state.hoveredTag} />}
                hoverable
                position="top center"
              />
            </div>
          </div>
        ))}
      </div>
    );
  }
}

export default Collections;
