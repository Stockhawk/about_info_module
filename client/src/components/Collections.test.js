import React from 'react';
import { shallow } from 'enzyme';

import Collections from './Collections';

describe('App', () => {
  it('should render correctly in debug mode', () => {
    const component = shallow(<Collections debug/>);
    expect(component).toMatchSnapshot();
  });
});
