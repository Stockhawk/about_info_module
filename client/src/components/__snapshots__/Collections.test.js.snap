// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly in debug mode 1`] = `
<div>
  <div>
    <h4>
      Collections
    </h4>
  </div>
  <div>
    <span>
      <button>
        Female CEOs
      </button>
      <button>
        100 Most Popular
      </button>
      <button>
        Manufacturing
      </button>
      <button>
        Electronics
      </button>
    </span>
  </div>
</div>
`;
